### PR TITLE
Feat/15406856 main extenstion page

### PR DESCRIPTION
**Pull request by CzarekDryl**
Added new main extension page, changed routing, added single extension component.

image
image

### CODE REVIEW

File: src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import ExtensionStatusBadge from '../ExtensionStatusBadge-new/ExtensionStatusBadge';
+import Button from '~shared/Extensions/Button';
+import Icon from '~shared/Icon';
+import { useAsyncFunction, useColonyContext, useExtensionData, useMobile } from '~hooks';
+import { ActionTypes } from '~redux';
+import Link from '~shared/Link/Link';
+import styles from './ExtensionItem.module.css';
+import { isInstalledExtensionData } from '~utils/extensions';
+import { ExtensionStatusBadgeMode } from '../ExtensionStatusBadge-new/types';
+
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx:13]: displayName missing

File: src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+const ExtensionItem: FC<ExtensionItemProps> = ({ title, description, version, icon, extensionId }) => {
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx:14]: extensionId = ""

File: src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  const { formatMessage } = useIntl();
+  const isMobile = useMobile();
+  const { colony } = useColonyContext();
+  const { extensionData } = useExtensionData(extensionId ?? '');
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx:18]: then  here just: `extensionId`

File: src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  const [status, setStatus] = useState<ExtensionStatusBadgeMode>();
+  const [badgeMessage, setBadgeMessage] = useState<string>('');
+
+  const values = useMemo(() => {
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx:22]: name `values` doesn't say anything, please change it

File: src/components/common/Extensions/Pages/Header/Header.module.css [hunk 5]
```diff
@@ -0,0 +1,3 @@
+.tooltipContainer {
+  @apply bg-white p-[1.2rem] shadow-default border border-gray-200 rounded-lg;
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/Pages/Header/Header.module.css:2]: bg-base-white

File: src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx (hunk 1 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx:13]: displayName added

File: src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx (hunk 2 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx]: Fixed

File: src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx (hunk 3 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx]: Fixed

File: src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx (hunk 4 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx]: changed to `extensionValues`

File: src/components/common/Extensions/Pages/Header/Header.module.css (hunk 5 above)
Comment (CzarekDryl) [reply src/components/common/Extensions/Pages/Header/Header.module.css]: Changed also in two different files

File: src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    };
+  }, [colony?.colonyAddress, extensionData]);
+
+  const submit = ActionTypes.EXTENSION_INSTALL;
+  const error = ActionTypes.EXTENSION_INSTALL_ERROR;
+  const success = ActionTypes.EXTENSION_INSTALL_SUCCESS;
+
+  const asyncFunction = useAsyncFunction({ submit, error, success });
+
+  const handleClick = useCallback(async () => {
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx:37]: `handleClick` not say what action you handle, please use better name

File: src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+import { ActionTypes } from '~redux';
+import Link from '~shared/Link/Link';
+import styles from './ExtensionItem.module.css';
+import { isInstalledExtensionData } from '~utils/extensions';
+import { ExtensionStatusBadgeMode } from '../ExtensionStatusBadge-new/types';
+
+const displayName = 'common.Extensions.ExtensionItem';
+
+const ExtensionItem: FC<ExtensionItemProps> = ({ title, description, version, icon, extensionId = '' }) => {
+  const { formatMessage } = useIntl();
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx:17]: Please move all logic to hooks.ts file

File: src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx [hunk 8]
```diff
@@ -0,0 +1,95 @@
+import React, { FC, useCallback, useMemo, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { ExtensionItemProps } from './types';
+import ExtensionStatusBadge from '../ExtensionStatusBadge-new/ExtensionStatusBadge';
+import Button from '~shared/Extensions/Button';
+import Icon from '~shared/Icon';
+import { useAsyncFunction, useColonyContext, useExtensionData, useMobile } from '~hooks';
+import { ActionTypes } from '~redux';
+import Link from '~shared/Link/Link';
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/ExtensionItem/ExtensionItem.tsx:9]: import Link from '~shared/Link';

File: src/components/common/Extensions/Pages/MainExtensionPage/MainExtensionPage.tsx [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    );
+  }
+
+  const allExtensions = [...availableExtensionsData, ...installedExtensionsData];
+
+  return (
+    <div className="mb-6">
+      {/* @TODO: Add sidepanel */}
+      {isMobile && <div>Sidepanel</div>}
+      <div className="mt-9 sm:mt-6">
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/Pages/MainExtensionPage/MainExtensionPage.tsx:31]: Please use <ThreeColumns />

**Review by krawczykkatarzyna — APPROVED** [review]

**Pull request merged**